File: backend_new/app/models/analysis_result.py
from sqlalchemy import Column, Integer, String, DateTime, Float, Boolean, ForeignKey, JSON
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.db.database import Base


class AnalysisResult(Base):
    __tablename__ = "analysis_results"

    id = Column(Integer, primary_key=True, index=True)
    call_session_id = Column(Integer, ForeignKey("call_sessions.id", ondelete="CASCADE"), nullable=False)
    chunk_index = Column(Integer, nullable=False)
    timestamp = Column(DateTime(timezone=True), server_default=func.now())
    risk_score = Column(Float, nullable=False)
    risk_level = Column(String, nullable=False)
    is_deepfake = Column(Boolean, default=False)
    confidence = Column(Float, nullable=False)
    transcript = Column(String, nullable=True)
    voice_characteristics = Column(JSON, nullable=True)

    # Relationships
    call_session = relationship("CallSession", back_populates="analysis_results")

File: ScamShieldAndroid/app/src/main/java/com/scamshield/domain/model/AnalysisResult.kt
package com.scamshield.domain.model

import java.time.LocalDateTime

data class AnalysisResult(
    val id: Long = 0,
    val callSessionId: Long,
    val chunkIndex: Int,
    val timestamp: LocalDateTime,
    val riskScore: Float,
    val riskLevel: RiskLevel,
    val isDeepfake: Boolean,
    val confidence: Float,
    val transcript: String? = null,
    val voiceCharacteristics: VoiceCharacteristics? = null
)

data class VoiceCharacteristics(
    val pitchMean: Float,
    val pitchStd: Float,
    val spectralCentroid: Float,
    val syntheticScore: Float
)
